tests/src/tests/unhappy.rs

```diff
@@ -44,15 +44,31 @@ async fn ten_nodes_one_down() {
         .then(|n| async move { n.consensus_info().await })
         .collect::<Vec<_>>()
         .await;
+
     // check that they have the same block
-    let blocks = infos
+    let target_blocks = infos
         .iter()
-        .map(|i| {
-            i.safe_blocks
-                .values()
-                .find(|b| b.view == TARGET_VIEW)
-                .unwrap()
-        })
+        .map(|i| i.safe_blocks.values().find(|b| b.view == TARGET_VIEW))
```

Comment:
nit: maybe a `flat_map` would be more convenient so that you have either an empty set or one with a block (and not a n `Option<block>`)

Reply:
I wanted to detect the false case that `Some<block>` from 8 nodes and `None` from 1 node, for example. If we use `flat_map` and the result contains only one block, we cannot probably be sure that all 9 nodes really have the block?

Reply:
Ah, I'm probably wrong. In unhappy, some nodes can have blocks and some other nodes can have timeout_qcs, in theory.

though, in the case I mentioned above, all nodes will have timeout_qc.
> this can sometimes happen if the dead node is selected as a leader who must propose a block for the view 20.

Also, this is what you suggested in https://github.com/logos-co/nomos-node/pull/438#discussion_r1336897533, right?

Reply:
Hm.. I'm confused. Let me think a bit.

Reply:
I think https://github.com/logos-co/nomos-node/pull/438#discussion_r1336911866 is correct :) That is, if some nodes have the TARGET_VIEW block, all other nodes must have the block. If some nodes have the TARGET_VIEW timeout_qc, other nodes shouldn't have the TARGET_VIEW block (because the Carnot architecture with deterministic random beacon doesn't allow the split-brain.). Did I understand it correctly?

Reply:
Yes exactly, if one agreement was reached for a view, either a node hasn't received it yet or it can only ever receive that same result. In the case of network splits Carnot will not make any progress

Reply:
Thanks. Then, I think we can't use `flat_map` because we can't probably detect the case where some nodes have the block 20 and others have the timeout_qc 20.